commit 25a3a701a4f0825a7b32adec97c85d6bf856ac39
Author: saahriktu <saahriktu@gmail.com>
Date:   Tue Apr 10 10:23:10 2018 +0300

    Добавление поддержки gopher-ссылок

diff --git a/src/main/java/ru/org/linux/util/URLUtil.java b/src/main/java/ru/org/linux/util/URLUtil.java
index f47ea7390..9eebb50ad 100644
--- a/src/main/java/ru/org/linux/util/URLUtil.java
+++ b/src/main/java/ru/org/linux/util/URLUtil.java
@@ -23,7 +23,7 @@ import java.util.regex.Pattern;
 
 public final class URLUtil {
   private static final Pattern isUrl = Pattern.compile(
-    "(((https?)|(ftp))://(([0-9\\p{L}.-]+\\.\\p{L}+)|(\\d+\\.\\d+\\.\\d+\\.\\d+))(:[0-9]+)?(/[^ ]*)?)|(mailto:[a-z0-9_+-.]+@[0-9a-z.-]+\\.[a-z]+)|(news:[a-z0-9.-]+)|(((www)|(ftp))\\.(([0-9a-z.-]+\\.[a-z]+(:[0-9]+)?(/[^ ]*)?)|([a-z]+(/[^ ]*)?)))",
+    "(((https?)|(ftp)|(gopher))://(([0-9\\p{L}.-]+\\.\\p{L}+)|(\\d+\\.\\d+\\.\\d+\\.\\d+))(:[0-9]+)?(/[^ ]*)?)|(mailto:[a-z0-9_+-.]+@[0-9a-z.-]+\\.[a-z]+)|(news:[a-z0-9.-]+)|(((www)|(ftp))\\.(([0-9a-z.-]+\\.[a-z]+(:[0-9]+)?(/[^ ]*)?)|([a-z]+(/[^ ]*)?)))",
     Pattern.CASE_INSENSITIVE
   );
 
